### PR TITLE
ioctl: nvme_io_args.dspec(Directive Specific) value is 16 bits

### PR DESCRIPTION
**Pull request by sc108-lee**
Command Dword 13 [31:16]

Signed-off-by: Steven Seungcheol Lee <sc108.lee@samsung.com>

### CODE REVIEW

File: src/nvme/ioctl.h [hunk 1]
```diff
@@ -3848,7 +3848,7 @@ struct nvme_io_args {
 	__u16 apptag;
 	__u16 appmask;
 	__u8 dsm;
-	__u8 dspec;
+	__u16 dspec;
```
Comment (igaw) [inline src/nvme/ioctl.h:3851]: Can you move the dspec before dsm so that we have all __u16 grouped together. This avoids holes in the data structures.

**Review by igaw — CHANGES_REQUESTED** [review]

File: src/nvme/ioctl.h (hunk 1 above)
Comment (sc108-lee) [reply src/nvme/ioctl.h:3850]: addressed :)

Comment (igaw) [reply src/nvme/ioctl.h:3850]: You need also to update all inline function in the header which use struct nvme_io_args. Older compilers are complaining if the order of the struct initialization are not strictly following the order of the deceleration.

File: doc/html/_sources/libnvme.rst.txt [hunk 2]
```diff
@@ -5245,8 +5245,8 @@ The nvme command status if a response was received (see
     __u16 control;
     __u16 apptag;
     __u16 appmask;
+    __u16 dspec;
     __u8 dsm;
-    __u8 dspec;
```
Comment (igaw) [inline doc/html/_sources/libnvme.rst.txt:5249]: BTW, could you drop the generated documentation update in this patch? This makes backporting unnecessary complex. And the plan is to update the generated documentation right before each release.

**Review by igaw — CHANGES_REQUESTED** [review]

File: doc/html/_sources/libnvme.rst.txt (hunk 2 above)
Comment (sc108-lee) [reply doc/html/_sources/libnvme.rst.txt:5249]: Thanks for letting me know, I will remove doc updates :)

**Comment by sc108-lee** [comment]
@igaw there is no direct use of this struct in inline function

**Comment by igaw** [comment]
Right you are! :)

**Pull request merged**